test/membrane_tcp/sink/pipeline_test.exs

```diff
@@ -0,0 +1,148 @@
+    Pipeline.terminate(pipeline)
+    Socket.close(socket)
+  end
+
+  defp receive_data(socket, received_data \\ "", terminator \\ ".") do
+    {:ok, payload} = Socket.recv(socket)
+
+    if String.ends_with?(payload, terminator) do
+      received_data <> payload
+    else
+      receive_data(socket, received_data <> payload)
+    end
+  end
+
+  defp create_connected_socket_pair() do
+    server_socket =
+      %Socket{connection_side: :server, ip_address: @localhost, port_no: @server_port_no}
+
+    client_socket =
+      %Socket{connection_side: :client, ip_address: @localhost, port_no: 0}
+
+    {:ok, listening_socket} = Socket.listen(server_socket)
+    {:ok, client_socket} = Socket.connect(client_socket, server_socket)
+    {:ok, server_socket} = Socket.accept(listening_socket)
+
+    Socket.close(listening_socket)
+
+    {client_socket, server_socket}
+  end
+
+  describe "100 messages pass through a" do
+    test "client-side pipeline created without a socket" do
+      assert {:ok, listening_socket} =
+               %Socket{
+                 connection_side: :server,
+                 ip_address: @localhost,
+                 port_no: @server_port_no,
+                 sock_opts: [active: false]
+               }
+               |> Socket.listen()
+
+      assert pipeline =
+               Pipeline.start_link_supervised!(
+                 spec:
+                   child(:source, %Source{output: @data})
+                   |> child(:sink, %Sink{
+                     local_address: @localhost,
+                     connection_side: {:client, @localhost, @server_port_no}
+                   }),
+                 test_process: self()
+               )
+
+      assert {:ok, socket} = Socket.accept(listening_socket)
+      Socket.close(listening_socket)
+      # time for a pipeline to enter playing playback
+      Process.sleep(200)
+      run_pipeline(pipeline, socket)
+    end
+
+    test "server-side pipeline created without a socket" do
+      assert pipeline =
+               Pipeline.start_link_supervised!(
+                 spec:
+                   child(:tcp_source, %Source{output: @data})
+                   |> child(:sink, %Sink{
+                     connection_side: :server,
+                     local_address: @localhost,
+                     local_port_no: @server_port_no
+                   }),
+                 test_process: self()
+               )
+
+      assert {:ok, connected_client_socket} =
+               %Socket{
+                 connection_side: {:client, @localhost, @server_port_no},
+                 port_no: 0,
+                 ip_address: @localhost,
+                 sock_opts: [active: false]
+               }
+               |> Socket.connect(%Socket{
+                 connection_side: :server,
+                 ip_address: @localhost,
+                 port_no: @server_port_no
+               })
+
+      # time for a pipeline to enter playing playback
+      Process.sleep(200)
```

Comment:
cannot you use `assert_sink_playing`?

Reply:
not really since I'm not using a testing source